ManiVault/src/plugins/ClusterData/src/ClusterData.cpp

```diff
@@ -61,6 +61,31 @@ void ClusterData::addCluster(Cluster& cluster)
     _clusters.push_back(cluster);
 }
 
+void ClusterData::setClusterNames(const std::vector<QString>& clusterNames)
+{
+    if (clusterNames.empty())
+        return;
+
+    if (clusterNames.size() != _clusters.size())
+    {
+        qWarning() << "ClusterData: Number of cluster names does not equal the number of data cluster. No cluster names assigned.";
+        return;
+    }
+
+    for (size_t i = 0; i < clusterNames.size(); i++)
+        _clusters[i].setName(clusterNames[i]);
+}
+
+std::vector<QString> ClusterData::getClusterNames()
+{
+    std::vector<QString> clusterNames;
+
```

Comment:
Just a small suggestion wrt optimization. Perhaps we could `clusterNames.reserve(_clusters.size());` here? When we have a large number of clusters this might speed up things a bit.

Reply:
That's a good idea!